src/main/java/com/mongodb/kafka/connect/util/config/ConfigSoftValidator.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.util.config;
+
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.TOPIC_OVERRIDE_PREFIX;
+import static com.mongodb.kafka.connect.util.Assertions.assertNotNull;
+import static com.mongodb.kafka.connect.util.Assertions.assertTrue;
+import static java.lang.String.format;
+import static java.util.Collections.unmodifiableSet;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.kafka.common.config.ConfigDef;
+
+import com.mongodb.annotations.Immutable;
+import com.mongodb.lang.Nullable;
+
+import com.mongodb.kafka.connect.sink.MongoSinkTopicConfig;
+
+/** Logs information about potential configuration issues. */
+public final class ConfigSoftValidator {
+  /**
+   * This method is used in our implementation of the {@link ConfigDef#validateAll(Map)} method
+   * because the original implementation silently disregards any undefined configuration properties.
+   *
+   * @param propertyNames See the parameter description in {@link #strippedPropertyName(String)}.
+   */
+  public static void logObsoleteProperties(
```

Comment:
The implementation of the `logObsoleteProperties` has changed from what we originally had in `SinkConfigSoftValidator` before its code was moved here. The `ObsoletePropertiesSet` class is also a new piece of code. The rest of the `ConfigSoftValidator` was just moved here from `SinkConfigSoftValidator`.